File: servers/game/src/main/kotlin/fr/rob/game/domain/player/Player.kt
package fr.rob.game.domain.player

import fr.rob.game.domain.entity.Unit
import fr.rob.game.domain.entity.guid.ObjectGuid
import fr.rob.game.domain.player.session.GameSession

class Player(
    val ownerGameSession: GameSession,
    guid: ObjectGuid,
    name: String,
    level: Int
) : Unit(guid, name, level) {

    init {
        controlledByGameSession = ownerGameSession
    }
}

File: RobClient/src/game/entity/Player.cs
using RobClient.Game.Entity.Guid;

namespace RobClient.Game.Entity;

public class Player : WorldObject
{
    public string Name
    { get; private set; }

    public Player(string name, ObjectGuid guid, Vector4f position) : base(guid, position)
    {
        Name = name;
    }
}